builtin/logical/transit/path_export_test.go

```diff
@@ -381,3 +392,134 @@ func TestTransit_Export_EncryptionKey_DoesNotExportHMACKey(t *testing.T) {
+	})
+	require.NoError(t, err)
+
+	testTransit_exportCertificateChain(t, client, "rsa-2048")
+	testTransit_exportCertificateChain(t, client, "rsa-3072")
+	testTransit_exportCertificateChain(t, client, "rsa-4096")
+	testTransit_exportCertificateChain(t, client, "ecdsa-p256")
+	testTransit_exportCertificateChain(t, client, "ecdsa-p384")
+	testTransit_exportCertificateChain(t, client, "ecdsa-p521")
+	testTransit_exportCertificateChain(t, client, "ed25519")
+}
+
+func testTransit_exportCertificateChain(t *testing.T, apiClient *api.Client, keyType string) {
+	keyName := fmt.Sprintf("%s", keyType)
+	issuerName := fmt.Sprintf("%s-issuer", keyType)
+
+	// Get key to be imported
+	privKey := getKey(t, keyType)
+	privKeyBytes, err := x509.MarshalPKCS8PrivateKey(privKey)
+	require.NoError(t, err, fmt.Sprintf("failed to marshal private key: %s", err))
+
+	// Create CSR
+	var csrTemplate x509.CertificateRequest
+	csrTemplate.Subject.CommonName = "example.com"
+	csrBytes, err := x509.CreateCertificateRequest(cryptoRand.Reader, &csrTemplate, privKey)
+	require.NoError(t, err, fmt.Sprintf("failed to create CSR: %s", err))
+
+	pemCsr := string(pem.EncodeToMemory(&pem.Block{
+		Type:  "CERTIFICATE REQUEST",
+		Bytes: csrBytes,
+	}))
+
+	// Generate PKI root
+	_, err = apiClient.Logical().Write("pki/root/generate/internal", map[string]interface{}{
+		"issuer_name": issuerName,
+		"common_name": "PKI Root X1",
+	})
+	require.NoError(t, err)
+
+	// Create role to be used in the certificate issuing
+	_, err = apiClient.Logical().Write("pki/roles/example-dot-com", map[string]interface{}{
+		"issuer_ref":                         issuerName,
+		"allowed_domains":                    "example.com",
+		"allow_bare_domains":                 true,
+		"basic_constraints_valid_for_non_ca": true,
+		"key_type":                           "any",
+	})
+	require.NoError(t, err)
+
+	// Sign the CSR
+	resp, err := apiClient.Logical().Write("pki/sign/example-dot-com", map[string]interface{}{
+		"issuer_ref": issuerName,
+		"csr":        pemCsr,
+		"ttl":        "10m",
+	})
+	require.NoError(t, err)
+	require.NotNil(t, resp)
+
+	leafCertPEM := resp.Data["certificate"].(string)
+
+	// Get wrapping key
+	resp, err = apiClient.Logical().Read("transit/wrapping_key")
+	require.NoError(t, err)
+	require.NotNil(t, resp)
+
+	pubWrappingKeyString := strings.TrimSpace(resp.Data["public_key"].(string))
+	wrappingKeyPemBlock, _ := pem.Decode([]byte(pubWrappingKeyString))
+
+	pubWrappingKey, err := x509.ParsePKIXPublicKey(wrappingKeyPemBlock.Bytes)
+	require.NoError(t, err, "failed to parse wrapping key")
+
+	blob := wrapTargetPKCS8ForImport(t, pubWrappingKey.(*rsa.PublicKey), privKeyBytes, "SHA256")
+
+	// Import key
+	_, err = apiClient.Logical().Write(fmt.Sprintf("/transit/keys/%s/import", keyName), map[string]interface{}{
+		"ciphertext": blob,
+		"type":       keyType,
+	})
+	require.NoError(t, err)
+
+	// NOTE: Should it be possible for the "import" endpoint to also import the cert chain?
```

Comment:
Nah, in general I'd probably expect them to be fine with it being two separate operations. The bulk majority of users probably want to generate a CSR/certificate from key material already inside Vault. :-)